integration-tests/smoke/ccip_batching_test.go

```diff
@@ -0,0 +1,295 @@
+		err := sendMessages(
+			ctx,
+			t,
+			e.Env.Chains[sourceChain1],
+			e.Env.Chains[sourceChain1].DeployerKey,
+			state.Chains[sourceChain1].OnRamp,
+			state.Chains[sourceChain1].Router,
+			state.Chains[sourceChain1].Multicall3,
+			destChain,
+			numMessages,
+			common.LeftPadBytes(state.Chains[destChain].Receiver.Address().Bytes(), 32),
+		)
+		require.NoError(t, err)
+
+		_, err = ccdeploy.ConfirmCommitWithExpectedSeqNumRange(
+			t,
+			e.Env.Chains[sourceChain1],
+			e.Env.Chains[destChain],
+			state.Chains[destChain].OffRamp,
+			nil,
+			ccipocr3.NewSeqNumRange(startSeqNum[sourceChain1], endSeqNum[sourceChain1]),
+		)
+		require.NoErrorf(t, err, "failed to confirm commit from chain %d", sourceChain1)
+
+		startSeqNum[sourceChain1] = endSeqNum[sourceChain1] + 1
+		endSeqNum[sourceChain1] = startSeqNum[sourceChain1] + ccipocr3.SeqNum(numMessages) - 1
+	})
+
+	t.Run("batch data only messages from multiple sources", func(t *testing.T) {
+		var (
+			wg   sync.WaitGroup
+			mx   sync.Mutex
+			errs []error
+		)
+
+		wg.Add(1)
+		go func(sourceChainSelector uint64) {
+			defer wg.Done()
+			err := sendMessages(
+				ctx,
+				t,
+				e.Env.Chains[sourceChainSelector],
+				e.Env.Chains[sourceChainSelector].DeployerKey,
+				state.Chains[sourceChainSelector].OnRamp,
+				state.Chains[sourceChainSelector].Router,
+				state.Chains[sourceChainSelector].Multicall3,
+				destChain,
+				numMessages,
+				common.LeftPadBytes(state.Chains[destChain].Receiver.Address().Bytes(), 32),
+			)
+			mx.Lock()
+			errs = append(errs, err)
+			mx.Unlock()
+		}(sourceChain1)
+
+		wg.Add(1)
+		go func(sourceChainSelector uint64) {
+			defer wg.Done()
+			err := sendMessages(
+				ctx,
+				t,
+				e.Env.Chains[sourceChainSelector],
+				e.Env.Chains[sourceChainSelector].DeployerKey,
+				state.Chains[sourceChainSelector].OnRamp,
+				state.Chains[sourceChainSelector].Router,
+				state.Chains[sourceChainSelector].Multicall3,
+				destChain,
+				numMessages,
+				common.LeftPadBytes(state.Chains[destChain].Receiver.Address().Bytes(), 32),
+			)
+			mx.Lock()
+			errs = append(errs, err)
+			mx.Unlock()
+		}(sourceChain2)
+
+		wg.Wait()
+
+		for _, err := range errs {
+			require.NoError(t, err)
+		}
+
+		// confirm the commit reports
+		var (
+			sourceChain1Report *offramp.OffRampCommitReportAccepted
+			sourceChain2Report *offramp.OffRampCommitReportAccepted
+		)
+		errs = nil
+
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			var err error
+			sourceChain1Report, err = ccdeploy.ConfirmCommitWithExpectedSeqNumRange(t,
+				e.Env.Chains[sourceChain1],
+				e.Env.Chains[destChain],
+				state.Chains[destChain].OffRamp,
+				nil,
+				ccipocr3.NewSeqNumRange(startSeqNum[sourceChain1], endSeqNum[sourceChain1]),
+			)
+
+			mx.Lock()
+			errs = append(errs, err)
+			mx.Unlock()
+		}()
+
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			var err error
+			sourceChain2Report, err = ccdeploy.ConfirmCommitWithExpectedSeqNumRange(t,
```

Comment:
Does this check enforce all seqNrs to be in the same root?

Reply:
Yeah, believe so, it checks that the root min seq nr >= the given min seq nr and max seq nr <= the given max seq nr

Reply:
I double checked and it does so 👍